### ISSUE TITLE
Rename Digest to Hash

### ISSUE THREAD

**Issue by cespare** [cespare/xxhash#48]
Digest is not a good name. Normally "digest" refers to the output of the hash function.

In v3 we could consider renaming it; perhaps to `Hash`.

Typical code probably doesn't refer to `Digest` by name and doesn't have to change:

```
d := xxhash.New()
... call d.Write() ...
x := d.Sum64()
```

**Comment by greatroar** [comment]
Hash also refers the output of a hash function :)

**Comment by cespare** [comment]
Yes, but it it is also the name of the type of function...as in **hash** function. (Plus this matches `hash.Hash`.)

Open to better suggestions though :)
